Source/Core/DolphinQt/Config/ConfigControls/ConfigChoice.cpp

```diff
@@ -61,8 +67,21 @@ void ConfigStringChoice::Connect()
   connect(this, &QComboBox::currentIndexChanged, this, &ConfigStringChoice::Update);
 }
 
+void ConfigStringChoice::Update(int index)
+{
+  const QVariant data_value = itemData(index);
+  const QString setting_value = data_value.isValid() ? data_value.toString() : itemText(index);
+  Config::SetBaseOrCurrent(m_setting, setting_value.toStdString());
+}
+
 void ConfigStringChoice::Load()
 {
+  const QString setting_value = QString::fromStdString(Config::Get(m_setting));
+
+  int index = findData(setting_value);
+  if (index == -1)
+    index = findText(setting_value);
```

Comment:
This strikes me as a bit odd... it might be better to store whether this `ConfigStringChoice` instance is a data-is-text or a data-is-separate style combobox and then only query the correct method here.